src/rt.rs

```diff
@@ -44,49 +97,60 @@ pub fn set_allowed_backends(backends: &str) {
 }
 
 pub fn get_program_class() -> Option<String> {
+    assert_initialized_main_thread!();
     unsafe {
         from_glib_none(ffi::gdk_get_program_class())
     }
 }
 
 pub fn set_program_class(program_class: &str) {
+    assert_initialized_main_thread!();
     unsafe {
         ffi::gdk_set_program_class(program_class.to_glib_none().0)
     }
 }
 
 pub fn flush() {
+    assert_initialized_main_thread!();
```

Comment:
This is exactly what I was afraid of. We now have to call the macro everywhere. :-/